app/components/QRCodeReader.js

```diff
@@ -17,18 +18,47 @@ export default class QRCodeReader extends React.Component {
       canvas.width = this.width;
       canvas.height = this.height;
       context.drawImage(this, 0, 0);
-      const imageData = context.getImageData(0, 0, this.width, this.height);
+
+      const convertToGrayScale = (imageData) => {
```

Comment:
This function should be moved to the root level of this file. Maybe at the bottom.